iconrpcserver/dispatcher/default/websocket.py

```diff
@@ -104,56 +105,50 @@ class WSDispatcher:
                 await ws.send(json.dumps(request))
                 heartbeat_time = StubCollection().conf[ConfigKey.WS_HEARTBEAT_TIME]
                 await asyncio.sleep(heartbeat_time)
+            except exceptions.ConnectionClosed:
+                Logger.debug("Connection Closed by child.")  # TODO: Useful message needed.
             except Exception as e:
-                exception = e
-                traceback.print_exc()
-                break
-
-        if not exception:
-            exception = ConnectionError("Connection closed.")
-
-        error_code = message_code.Response.fail_connection_closed
-        await WSDispatcher.send_and_raise_exception(ws, "node_ws_PublishHeartbeat", exception, error_code)
+                traceback.print_exc()  # TODO: Keep this tb?
+                await WSDispatcher.send_exception(
+                    ws, call_method,
+                    exception=e,
+                    error_code=message_code.Response.fail_connection_closed
+                )
 
     @staticmethod
     async def publish_new_block(ws, channel_name, height, peer_id):
-        exception = None
-        error_code = None
+        call_method = "node_ws_PublishNewBlock"
         channel_stub = get_channel_stub_by_channel_name(channel_name)
         try:
-            while ws.open:
+            while True:
                 new_block_dumped, confirm_info_bytes = await \
                     channel_stub.async_task().announce_new_block(subscriber_block_height=height, subscriber_id=peer_id)
                 new_block: dict = json.loads(new_block_dumped)
                 confirm_info = confirm_info_bytes.decode('utf-8')
-                request = Request("node_ws_PublishNewBlock", block=new_block, confirm_info=confirm_info)
-                Logger.debug(f"node_ws_PublishNewBlock: {request}")
+                request = Request(call_method, block=new_block, confirm_info=confirm_info)
+                Logger.debug(f"{call_method}: {request}")
 
                 await ws.send(json.dumps(request))
                 height += 1
-        except exceptions.ConnectionClosed as e:
-            exception = e
-            error_code = message_code.Response.fail_connection_closed
+        except exceptions.ConnectionClosed:
+            Logger.debug("Connection Closed by child.")  # TODO: Useful message needed.
         except Exception as e:
-            exception = e
-            error_code = message_code.Response.fail_announce_block
-            traceback.print_exc()
-
-        if not exception:
-            exception = ConnectionError("Connection closed.")
-
-        await WSDispatcher.send_and_raise_exception(ws, "node_ws_PublishNewBlock", exception, error_code)
+            traceback.print_exc()  # TODO: Keep this tb?
+            await WSDispatcher.send_exception(
+                ws, call_method,
+                exception=e,
+                error_code=message_code.Response.fail_announce_block
+            )
```

Comment:
Need to break and return this loop as well.

Reply:
https://github.com/icon-project/icon-rpc-server/pull/106#discussion_r336318411

Reply:
https://github.com/icon-project/icon-rpc-server/pull/106#discussion_r336323661